ulib/axstarry/src/syscall_fs/ctype/eventfd.rs

```diff
@@ -1,29 +1,48 @@
+use alloc::sync::Arc;
 use axerrno::AxResult;
 use axfs::api::{FileIO, FileIOType};
+use axsync::Mutex;
 
 pub struct EventFd {
-    value: u32,
+    value: Arc<Mutex<u64>>,
     flags: i32,
 }
 
 impl EventFd {
-    pub fn new(initval: u32, flags: i32) -> EventFd {
+    pub fn new(initval: u64, flags: i32) -> EventFd {
         EventFd {
-            value: initval,
+            value: Arc::new(Mutex::new(initval)),
             flags,
         }
     }
 }
 
 impl FileIO for EventFd {
     fn read(&self, buf: &mut [u8]) -> AxResult<usize> {
-        let len: usize = core::mem::size_of::<u32>();
+        let len: usize = core::mem::size_of::<u64>();
         assert!(buf.len() == len);
 
-        buf[0..len].copy_from_slice(&self.value.to_ne_bytes());
+        buf[0..len].copy_from_slice(&self.value.lock().to_ne_bytes());
         Ok(len)
     }
 
+    fn write(&self, buf: &[u8]) -> AxResult<usize> {
+        let len: usize = core::mem::size_of::<u64>();
+        assert!(buf.len() == len);
+
+        let val = u64::from_ne_bytes(buf[0..len].try_into().unwrap());
+        let mut value_guard = self.value.lock();
+        match value_guard.checked_add(val) {
+            Some(new_value) => {
+                *value_guard = new_value;
```

Comment:
The eventfd is defined here: [https://man7.org/linux/man-pages/man2/eventfd2.2.html](url).

The original definition of the write operation is as follows:

> A [write(2)](https://man7.org/linux/man-pages/man2/write.2.html) call adds the 8-byte integer value supplied in
 its buffer to the counter.  The maximum value that may be
 stored in the counter is the largest unsigned 64-bit value
 minus 1 (i.e., 0xfffffffffffffffe).

When `EFD_SEMAPHORE` flag is not set for the eventfd, it cannot use an assignment instead of an addition.

Reply:
Yes, you are right. I will handle flag in future MR. As of now, the read/write implementation is very basic.

Reply:
> adds the 8-byte integer value

The spec says 8-byte, but I am not sure whether I should use `usize` or `u64`. Maybe `usize` is more versatile? (P.S. I changed from usize to u64 in this MR)

Reply:
It should use the `u64`, cause the documentation clearly states that it is 64-bit and use the `uint_64` type. The `usize` should be used when the argument use the type like `size_t`